### PR TITLE
Fixes/room service payloads

### PR DESCRIPTION
**Pull request by imxm**
## Random changes and fixes
- me response changed
- pitch slot response changed
- query params updated to support filtering for my rooms only
- user edit endpoint fixed
- company creation on gmail signup fixed

### CODE REVIEW

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -90,6 +96,19 @@ export class AuthService {
...
+        appType: newUser.twitter ? 'twitter' : null,
+        profileLink: newUser.twitter,
+      },
+      {
+        appType: newUser.linkedIn ? 'linkedIn' : null,
+        profileLink: newUser.linkedIn,
+      },
+    ] as SocialSites[]).filter(socialSite => socialSite.appType);
+    if (!newUser['socialSites']?.length) {
+      delete newUser['socialSites'];
```
Comment (masoodtalha) [inline src/auth/auth.service.ts:110]: @imxm No magic strings please

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -124,7 +143,26 @@ export class AuthService {
       }
       throw new InternalServerErrorException(defaultInternalServerErrorResponse);
     } else {
-      return await this.usersService.insertOne(newUser);
+
+      let session;
+      try {
+        session = await this.companyService.getSession();
+        await session.startTransaction();
+        const insertedUser = await this.usersService.insertOne(newUser, { session });
```
Comment (masoodtalha) [inline src/auth/auth.service.ts:151]: @imxm type?

File: src/auth/auth.service.ts [hunk 3]
```diff
@@ -321,11 +362,23 @@ export class AuthService {
               },
             ],
           };
-          return await this.usersService.insertOne(user);
+          session = await this.companyService.getSession();
+          await session.startTransaction();
+          const newUser = await this.usersService.insertOne(user, { session });
+            const companyMembersShip = this.compnayHelper.createCompanyMembersShip(newUser, type);
```
Comment (masoodtalha) [inline src/auth/auth.service.ts:368]: Types?

File: src/helpers/iCal.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    this.addToBody('PRODID:-//' + icsParams.organizerName + '//' + icsParams.productName + '//' + icsParams.language);
+    this.addToBody('X-WR-CALNAME:test');
+    this.addToBody('NAME:test');
+    this.addToBody('CALSCALE:GREGORIAN');
+    this.addToBody('METHOD:REQUEST');
+    this.addToBody('BEGIN:VTIMEZONE');
+    this.addToBody('TZID:' + icsParams.timezone);
+    this.addToBody('END:VTIMEZONE');
+    this.addToBody('BEGIN:VEVENT');
+    this.addToBody('UID:' + icsParams.UID);
```
Comment (masoodtalha) [inline src/helpers/iCal.ts:116]: @imxm Didn't we agree to remove this all?

File: src/auth/auth.service.ts (hunk 1 above)
Comment (imxm) [reply src/auth/auth.service.ts:110]: fixed, socialSites in `appType`. but the key name `newUser'[socialSite']` is needed for db.

File: src/auth/auth.service.ts (hunk 2 above)
Comment (imxm) [reply src/auth/auth.service.ts:151]: added types

File: src/auth/auth.service.ts (hunk 3 above)
Comment (imxm) [reply src/auth/auth.service.ts:368]: fixed

File: src/helpers/iCal.ts (hunk 4 above)
Comment (imxm) [reply src/helpers/iCal.ts:116]: yes we agreed, but in next sprint i will fix this and other calendar invite related issues, like organizer name in gmail. but can be changed now if you say so, but will take some time to research

**Pull request merged**